tests/operators/test_grid_sampling_op.py

```diff
@@ -0,0 +1,268 @@
+"""Tests for sensitivity operator."""
+
+# Copyright 2024 Physikalisch-Technische Bundesanstalt
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#       http://www.apache.org/licenses/LICENSE-2.0
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+import pytest
+import torch
+from mrpro.data import SpatialDimension
+from mrpro.operators import GridSamplingOp
+from torch.autograd.gradcheck import gradcheck
+
+from tests import RandomGenerator
+from tests.helper import dotproduct_adjointness_test
+
+
+@pytest.mark.parametrize('dtype', ['float32', 'float64', 'complex64'])
+def test_grid_sampling_op_dtype(dtype):
+    """Test for different data types."""
+    _test_grid_sampling_op_adjoint(dtype=dtype)
+
+
+@pytest.mark.parametrize('dim_str', ['2D', '3D'])
+@pytest.mark.parametrize('batched', ['batched', 'non_batched'])
+@pytest.mark.parametrize('channel', ['multi_channel', 'single_channel'])
+def test_grid_sampling_op_dim_batch_channel(dim_str, batched, channel):
+    """Test for different dimensions."""
+    _test_grid_sampling_op_adjoint(dim=int(dim_str[0]), batched=batched, channel=channel)
+
+
+@pytest.mark.parametrize('interpolation_mode', ['bilinear', 'nearest', 'bicubic'])
+def test_grid_sampling_op_interpolation_mode(interpolation_mode):
+    """Test for different interpolation_modes."""
+    _test_grid_sampling_op_adjoint(dim=2, interpolation_mode=interpolation_mode)
+
+
+@pytest.mark.parametrize('padding_mode', ['zeros', 'border', 'reflection'])
+def test_grid_sampling_op_padding_mode(padding_mode):
+    """Test for different padding_modes."""
+    _test_grid_sampling_op_adjoint(padding_mode=padding_mode)
+
+
+@pytest.mark.parametrize('align_corners', ['no_align', 'align'])
+def test_grid_sampling_op_align_mode(align_corners):
+    """Test for different align modes ."""
+    _test_grid_sampling_op_adjoint(align_corners=align_corners == 'align')
```

Comment:
should this not be `align_corners=align_corners` as align_corners is a string and not bool